akka-projection-cassandra/src/main/scala/akka/projection/cassandra/internal/CassandraProjectionImpl.scala

```diff
@@ -81,11 +83,34 @@ import akka.stream.scaladsl.Source
   private val promiseToStop: Promise[Done] = Promise()
   private val started = new AtomicBoolean(false)
 
+  override def withSettings(projectionSettings: ProjectionSettings): Projection[Envelope] = {
+    new CassandraProjectionImpl(projectionId, sourceProvider, strategy, Option(projectionSettings), handler)
+  }
+
   override def run()(implicit systemProvider: ClassicActorSystemProvider): Unit = {
     if (started.compareAndSet(false, true)) {
       implicit val system: ActorSystem = systemProvider.classicSystem
+      val done = mappedSource().run()
+      promiseToStop.completeWith(done)
+    }
+  }
 
-      val done = mappedSource().runWith(Sink.ignore)
+  override def runWithBackoff()(implicit systemProvider: ClassicActorSystemProvider): Unit = {
+
+    val projectionSettings = projectionSettingsOpt.getOrElse(ProjectionSettings(systemProvider))
+
+    if (started.compareAndSet(false, true)) {
+      implicit val system: ActorSystem = systemProvider.classicSystem
+      val done =
+        RestartSource
+          .withBackoff(
+            projectionSettings.minBackoff,
+            projectionSettings.maxBackoff,
+            projectionSettings.randomBackoffFactor) { () =>
+            mappedSource()
+          }
+          .via(killSwitch.flow)
```

Comment:
I'm not sure if that's the best way of doing that. The killSwitch is used in two places: in the restart source, so we can stop the whole projection and in the underlying `mappedSource`. This is useful when running from the testkit, ie: when using `run()` or `mappedSource()`